bulbs/poll/models.py

```diff
@@ -0,0 +1,94 @@
+from bulbs.content.models import Content
+from bulbs.utils import vault
+from django.db import models
+from djes.models import Indexable
+from rest_framework.exceptions import APIException
+
+from time import mktime
+import pytz
+import requests
+
+SODAHEAD_DATE_FORMAT = '%m/%d/%y %I:%M %p'
+
+class Poll(Content):
+    class SodaheadResponseError(APIException):
+        status_code = 503
+        default_detail = "Third-party poll provider temporarily unavailable."
+
+    question_text = models.TextField(blank=True, default="")
+    sodahead_id = models.CharField(max_length=20, blank=True, default="")
+    last_answer_index = models.IntegerField(default=0)
+    end_date = models.DateTimeField(null=True, default=None)
+
+    def sodahead_payload(self):
+        poll_payload = {
+            'access_token': vault.read('sodahead/token'),
+            'id': self.sodahead_id,
+            'name': self.title,
+            'title': self.question_text,
+        }
+
+        if self.published:
+            activation_date = self.published.astimezone(pytz.utc)
+            poll_payload['activationDate'] = activation_date.strftime(SODAHEAD_DATE_FORMAT)
+
+        if self.end_date:
+            end_date = self.end_date.astimezone(pytz.utc)
+            poll_payload['endDate'] = end_date.strftime(SODAHEAD_DATE_FORMAT)
+
+        for answer in self.answers.all():
+            if answer.answer_text is u'':
+                poll_payload[answer.sodahead_answer_id] = 'Intentionally blank'
+            else:
+                poll_payload[answer.sodahead_answer_id] = answer.answer_text
+
+        if not 'answer_01' in poll_payload:
+            poll_payload['answer_01'] = 'default answer 1'
+
+        if not 'answer_02' in poll_payload:
+            poll_payload['answer_02'] = 'default answer 2'
+
+        return poll_payload
+
+    def save(self, *args, **kwargs):
+        if self.sodahead_id is "":
+            response = requests.post('https://onion.sodahead.com/api/polls/', self.sodahead_payload())
+            if response.status_code is not 200:
+                raise Poll.SodaheadResponseError(response.text)
+            else:
+                self.sodahead_id = response.json()['poll']['id']
+        else:
+            self.sync_sodahead()
+
+        super(Poll, self).save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        response = requests.delete('https://onion.sodahead.com/api/polls/{}/?access_token={}'
+                .format(self.sodahead_id, vault.read('sodahead/token')))
+        if response.status_code is not 204:
+            raise Poll.SodaheadResponseError(response.text)
+        super(Poll, self).delete(*args, **kwargs)
+
+    def sync_sodahead(self):
+        response = requests.post('https://onion.sodahead.com/api/polls/{}/'
+                .format(self.sodahead_id), self.sodahead_payload())
+        if response.status_code is not 200:
```

Comment:
Good question. A heavyweight linter like "PyLint" would, not sure about
flake8.



On Mon, Jan 25, 2016 at 10:36 AM, collin <notifications@github.com> wrote:

> In bulbs/poll/models.py
> <https://github.com/theonion/django-bulbs/pull/96#discussion_r50718357>:
>
> > +        else:
> > +            self.sync_sodahead()
> > +
> > +        super(Poll, self).save(*args, **kwargs)
> > +
> > +    def delete(self, *args, **kwargs):
> > +        response = requests.delete('https://onion.sodahead.com/api/polls/{}/?access_token={}'
> > +                .format(self.sodahead_id, vault.read('sodahead/token')))
> > +        if response.status_code is not 204:
> > +            raise Poll.SodaheadResponseError(response.text)
> > +        super(Poll, self).delete(*args, **kwargs)
> > +
> > +    def sync_sodahead(self):
> > +        response = requests.post('https://onion.sodahead.com/api/polls/{}/'
> > +                .format(self.sodahead_id), self.sodahead_payload())
> > +        if response.status_code is not 200:
>
> [image: :+1:] will the linter suggest these changes too?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/theonion/django-bulbs/pull/96/files#r50718357>.
>



-- 
Mike Parent
Developer
Onion, Inc.